src/variant.h

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VARIANT_H
+#define BITCOIN_VARIANT_H
+
+#include <utility>
+
+#include <boost/variant.hpp>
+
+//! Substitute for C++17 std::variant
+template <class... Types>
+class Variant
+{
+private:
+    boost::variant<Types...> m_var;
+
+public:
+    Variant()
+    {
+        m_var = {};
+    }
+    Variant(const Variant& other)
+    {
+        m_var = other.m_var;
+    }
+    Variant(Variant& other)
+    {
+        m_var = other.m_var;
+    }
+    Variant(Variant&& other)
+    {
+        m_var = std::move(other.m_var);
+    }
+    template <class T>
+    Variant(const T& value)
+    {
+        m_var = value;
+    }
```

Comment:
In commit "Add a new Variant type" (85882c01d20add49f7b1f8b741e078968ec8080b)

I think this overload is also not doing anything

Reply:
Hmm seems that boost::variant implements this but std::variant doesn't.
doesn't this act as a copy constructor for a T?

Reply:
> Hmm seems that boost::variant implements this but std::variant doesn't.
> doesn't this act as a copy constructor for a T?

I think boost variant might implement this to deal with pre-c++11 compilers that don't support rvalue references. But the overload below `template <class T> Variant(T&& value)` should make this redundant with a c++11 compiler.

Reply:
Done